### PR TITLE
Fast Fourier transform

### PR DESCRIPTION
**Pull request by vladasp**
Issue #53 
WinForm application primary function transformation with Fast Fourier transformation.

### CODE REVIEW

**Comment by DmitryVSidorov** [comment]
description and reference to the corespondet issue should be added

File: src/Fourier transform/Fourier transform/FFT.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Fourier_transform
+{
+    public class FFT
+    {
+        /// <summary>
+        /// Вычисление поворачивающего модуля e^(-i*2*PI*k/N)
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/FFT.cs]: Only English comments

File: src/Fourier transform/Fourier transform/FFT.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        /// </summary>
+        /// <param name="k"></param>
+        /// <param name="N"></param>
+        /// <returns></returns>
+        private  Complex w(int k, int N)
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/FFT.cs]: what means name "w"

File: src/Fourier transform/Fourier transform/FFT.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Numerics;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Fourier_transform
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/FFT.cs]: please adhere to coding standards: "Fourier_transform"

File: src/Fourier transform/Fourier transform/Form1.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+
+namespace Fourier_transform
+{
+    
+    public partial class Form1 : Form
+    {
+        
+        string reference;
+        public Form1()
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/Form1.cs]: "Form1"

File: src/Fourier transform/Fourier transform/Form1.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        {
+            InitializeComponent();
+            chart1.Series.Add("f(x)");
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/Form1.cs]: "chart1"

File: src/Fourier transform/Fourier transform/Form1.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            chart1.Series["f(x)"].ChartType = System.Windows.Forms.DataVisualization.Charting.SeriesChartType.Line;
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/Form1.cs]: magic string "f(x)" and "F(x)"

File: src/Fourier transform/Fourier transform/Form1.cs [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            chart1.Series["f(x)"].Color = Color.DarkRed;
+            chart1.Series["f(x)"].BorderWidth = 7;
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/Form1.cs]: 7?

File: src/Fourier transform/Fourier transform/Form1.cs [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            reference = openTxtFileDialog.FileName;
+            textBox1.Text = reference;
+            string[] lines = File.ReadAllLines(reference);
+            foreach (string line in lines)
+            {
+                chart1.Series["f(x)"].Points.Add(Convert.ToDouble(line));
+            }
+        }
+
+        private void button2_Click(object sender, EventArgs e)
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/Form1.cs]: button2

File: src/Fourier transform/Fourier transform/rectangle.txt [hunk 9]
```diff
@@ -0,0 +1,256 @@
+0
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/rectangle.txt]: do not add this file

File: src/Fourier transform/Fourier transform/sinus.txt [hunk 10]
```diff
@@ -0,0 +1,256 @@
+0
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/sinus.txt]: do not add this file

File: src/Fourier transform/Fourier transform/triangle.txt [hunk 11]
```diff
@@ -0,0 +1,256 @@
+1
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/triangle.txt]: do not add this file

File: src/Fourier transform/Fourier transform/Form1.cs [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+            chart1.Series["f(x)"].ChartType = System.Windows.Forms.DataVisualization.Charting.SeriesChartType.Line;
+            chart1.Series["f(x)"].Color = Color.DarkRed;
+            chart1.Series["f(x)"].BorderWidth = 7;
+            chart2.Series.Add("F(x)");
+            chart2.Series["F(x)"].ChartType = System.Windows.Forms.DataVisualization.Charting.SeriesChartType.Line;
+            chart2.Series["F(x)"].Color = Color.DarkRed;
+            chart2.Series["F(x)"].BorderWidth = 7;
+        }
+
+        private void button1_Click(object sender, EventArgs e)
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/Form1.cs]: button1

File: src/Fourier transform/Fourier transform/Form1.cs [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+        {
+            if (chart1.Series["f(x)"].Points.Count != 0) chart1.Series["f(x)"].Points.Clear();
+            if (chart2.Series["F(x)"].Points.Count != 0) chart2.Series["F(x)"].Points.Clear();
+            openTxtFileDialog.Filter = "Txt|*.txt";
+            openTxtFileDialog.ShowDialog();
+            reference = openTxtFileDialog.FileName;
+            textBox1.Text = reference;
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/Form1.cs]: textBox1

File: src/Fourier transform/Fourier transform/Form1.cs [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+
+        private void button2_Click(object sender, EventArgs e)
+        {
+            string[] lines = File.ReadAllLines(reference);
+            Complex[] dataInputComplex = new Complex[lines.Length];
+            for (int c = 0; c < dataInputComplex.Length; c++)
+            {
+                dataInputComplex[c] = new Complex(c, Convert.ToDouble(lines[c]));
+            }
+            FFT Fft = new FFT();
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/Form1.cs]: the name "Fft" very bad for type FFT

File: src/Fourier transform/Fourier transform/FFT.cs [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+{
+    public class FFT
```
Comment (DmitryVSidorov) [inline src/Fourier transform/Fourier transform/FFT.cs]: is it really FFT must be a class???

File: src/Fourier transform/Fourier transform/FFT.cs (hunk 2 above)
Comment (vladasp) [inline src/Fourier transform/Fourier transform/FFT.cs]: "w" is standart designation of turning module

**Pull request merged**